sdk/src/Core/Amazon.Runtime/ClientConfig.cs

```diff
@@ -1115,6 +1082,17 @@ private static void ValidateTcpKeepAliveTimeSpan(TimeSpan? value, string paramNa
                 : (clientTimeout.HasValue ? clientTimeout : null);
         }
 
+        /// <summary>
+        /// Returns the endpoint that will be used for a particular request.
+        /// </summary>
+        /// <param name="parameters">A Container class for parameters used for endpoint resolution.</param>
+        /// <returns>The resolved endpoint for the given request.</returns>
+        public virtual Endpoint DetermineServiceOperationEndpoint(ServiceOperationEndpointParameters parameters)
```

Comment:
Seems like if this is ever called the null will cause a NullReferenceException. Should this method actually be `abstract`? Knowing this method will cause a failure in the request pipeline if called if there is a reason for it to not be `abstract` then maybe the base method should throw a meaningful error message.

Reply:
I don't think the request pipeline would call it considering how the generator is written. I just wanted to follow the same structure that we have in `BaseEndpointResolver.cs` https://github.com/aws/aws-sdk-net/blob/043879987f2d23429b15ceec73d8979c258e731a/sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseEndpointResolver.cs#L261
But I guess making it `abstract` would be reasonable.